keadm/cmd/keadm/app/cmd/util/common.go

```diff
@@ -166,15 +166,22 @@ func RunningModule() (types.ModuleRunning, error) {
 
 // GetLatestVersion return the latest non-prerelease, non-draft version of kubeedge in releases
 func GetLatestVersion() (string, error) {
-	//Download the tar from repo
-	versionURL := "curl -k " + latestReleaseVersionURL
-	cmd := exec.Command("sh", "-c", versionURL)
-	latestReleaseData, err := cmd.Output()
+	// curl https://kubeedge.io/latestversion
+	resp, err := http.Get(latestReleaseVersionURL)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to get latest version from %s: %v", latestReleaseVersionURL, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return "", err
+		}
+		return string(body), nil
 	}
 
-	return string(latestReleaseData), nil
+	return "", fmt.Errorf("failed to get latest version from %s, expected %d, got status code: %d", latestReleaseVersionURL, http.StatusOK, resp.StatusCode)
```

Comment:
/lgtm
nit: whether better adjust the sequence of two blocks, just like below:
```
if resp.StatusCode != http.StatusOK {
    return "", fmt.Errorf("failed to get latest version from %s, expected %d, got status code: %d", latestReleaseVersionURL, http.StatusOK, resp.StatusCode)
}

body, err := ioutil.ReadAll(resp.Body)
if err != nil {
    return "", err
}
return string(body), nil
```

/assign @fisherxu final decision

Reply:
Agree

Reply:
Sure! Easy to change 😄